conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureBodySerDe.java

```diff
@@ -65,12 +67,22 @@
 
     @Override
     public <T> Serializer<T> serializer(TypeMarker<T> token) {
-        return new EncodingSerializerRegistry<>(encodings, token);
+        return new EncodingSerializerRegistry<>(encodings, token, Optional.empty());
+    }
+
+    @Override
+    public <T> Serializer<T> serializer(TypeMarker<T> token, Endpoint endpoint) {
+        return new EncodingSerializerRegistry<>(encodings, token, Optional.of(endpoint));
     }
 
     @Override
     public <T> Deserializer<T> deserializer(TypeMarker<T> token) {
-        return new EncodingDeserializerRegistry<>(encodings, token);
+        return new EncodingDeserializerRegistry<>(encodings, token, Optional.empty());
+    }
+
+    @Override
+    public <T> Deserializer<T> deserializer(TypeMarker<T> token, Endpoint endpoint) {
+        return new EncodingDeserializerRegistry<>(encodings, token, Optional.of(endpoint));
```

Comment:
Was trying out if it reads nicer of we overload this constructor to avoid the optional wrapping here. However reverted this because it imo makes the flow harder to read and the `EncodingDeserializerRegistry` api is private anyway.